### ISSUE TITLE
Remove use of innerHtml

### ISSUE THREAD

**Issue by psiinon** [zaproxy/zap-hud#621]
Re https://github.com/zaproxy/zap-hud/pull/620

- [ ] zap-hud/src/main/zapHomeFiles/hud/display.js
  * Tricky - we are injecting the HTML ZAP returns - this will require significant rework
- [ ] zap-hud/src/main/zapHomeFiles/hud/libraries/alertify.js
  * This is a library outside of our control, but one we would like to replace: #306
- [ ] zap-hud/src/main/zapHomeFiles/hud/target/inject.js
  * This should be an easy fix

**Comment by mozfreddyb** [comment]
> Re #620
>     * [ ]  zap-hud/src/main/zapHomeFiles/hud/display.js  
>       * Tricky - we are injecting the HTML ZAP returns - this will require significant rework

If you're not against introducing a sanitizer (like DOMPurify), you could change this from
`                                window.open('').document.body.innerHTML = event.data.response;`
to
`                                window.open('').document.body.innerHTML = purify(event.data.response);`

and have the linter explicitly allow innerHTML as long as it's used in combination with dompurify.

**Comment by psiinon** [comment]
I have no problem adding DOMPurify to the ZAP domain - it could definitely prove useful. We're trying to limit what we include on the target domain, but that doesnt apply in this case.

**Comment by mozfreddyb** [comment]
if I were to add dompurify as a dependency in package.json - do you have a build step to move a copy to `src/main/zapHomeFiles/hud/libraries`?

**Comment by mozfreddyb** [comment]
Ah, I did not see you self-assigned. Nevermind then :-) I'll modify #620 to have the linter accept dompurify as an exception

**Comment by psiinon** [comment]
Yes, the HUD runtime JS is delivered by ZAP so we need it there.
And please got ahead with that change - I was going to look at the 3rd one first :)

**Comment by thc202** [comment]
> do you have a build step to move a copy to src/main/zapHomeFiles/hud/libraries?

Which ones are copied are tracked here:
https://github.com/zaproxy/zap-hud/blob/c1839b185616f673267e0cba5d46b9195a7a2d72/build.gradle.kts#L96-L106
